tests/keras/engine/test_training.py

```diff
@@ -1,17 +1,22 @@
+import os
 import pytest
 import numpy as np
 from numpy.testing import assert_allclose
 import sys
 import scipy.sparse as sparse
 
-from keras.layers import Dense, Dropout
+from keras.layers import Dense
+from keras.layers import Conv2D
+from keras.layers import Dropout
+from keras.layers import Flatten
 from keras.engine.topology import Input
 from keras.engine.training import Model, _check_loss_and_target_compatibility
 from keras.models import Sequential
 from keras import backend as K
 from keras.utils import Sequence
 from keras.utils.test_utils import keras_test
 from keras.callbacks import LambdaCallback
+from keras.callbacks import TensorBoard
```

Comment:
The callback instance should only be created in the tf case. Import shouldn't otherwise affect things. Moving might affect pep8